athenz/assertion.go

```diff
@@ -90,11 +171,88 @@ func resourceAssertionSchema() *schema.Schema {
 					Optional: true,
 					Default:  false,
 				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"condition": {
+					Type:     schema.TypeSet,
+					MaxItems: 2, // each assertion represent acl policy. Since in a given service,
+					// you can make the acl policy enforced on some hosts and not on others,
+					// therefor, to apply more than 2 conditions per assertion doesn't make sense
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"id": {
+								Computed: true,
+								Type:     schema.TypeInt,
+							},
+							"instances": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(nil),
+							},
+							"enforcementstate": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem: resourceConditionSchema(
+									validation.ToDiagFunc(
+										validation.StringInSlice([]string{"report", "enforce"}, false)),
+								),
+							},
+							"scopeonprem": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
+							},
+							"scopeaws": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
+							},
+							"scopeall": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
+							},
+						},
+					},
+				},
 			},
 		},
 	}
 }
 
+func expandAssertionConditions(configured []interface{}) *zms.AssertionConditions {
+	conditionsList := make([]*zms.AssertionCondition, 0, len(configured))
+	keys := []string{Instances, EnforcementState, ScopeONPREM, ScopeAWS, ScopeALL}
```

Comment:
what are these keys?

Reply:
Now I see its definition: `assertion conditions data keys`
But my question is- the order is matter ?

Reply:
no. I just you it in line 237 un order not to repeat myself...